xl-v1-common.go

```diff
@@ -0,0 +1,157 @@
+/*
+ * Minio Cloud Storage, (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	slashpath "path"
+	"path/filepath"
+)
+
+// Returns slice of disks needed for ReadFile operation:
+// - slice returing readable disks.
+// - fileMetadata
+// - bool value indicating if selfHeal is needed.
+// - error if any.
+func (xl XL) getReadableDisks(volume, path string) ([]StorageAPI, fileMetadata, bool, error) {
+	partsMetadata, errs := xl.getPartsMetadata(volume, path)
```

Comment:
Use variable declaration to make return variables more readable. listOnlineDisks sounds better.